rust/capture/src/router.rs

```diff
@@ -266,8 +269,22 @@ pub fn router<
         router = router.layer(ConcurrencyLimitLayer::new(limit));
     }
 
+    // add this prior to timeout middleware to ensure healthchecks are sensitive to load
+    router = router.merge(status_router);
+
+    if let Some(request_timeout_seconds) = request_timeout_seconds {
+        let timeout_duration = StdDuration::from_secs(request_timeout_seconds);
+        router = router.layer(axum::middleware::from_fn(
+            move |req: axum::extract::Request, next: axum::middleware::Next| async move {
+                match tokio::time::timeout(timeout_duration, next.run(req)).await {
+                    Ok(response) => response,
+                    Err(_) => (StatusCode::REQUEST_TIMEOUT, "Request timeout").into_response(),
+                }
+            },
+        ));
+    }
```

Comment:
suggestion: can we break it out to a separate function so that we don't duplicate the code between the runtime and tests?

Reply:
I was also thinking that it would be good to have a test that checks that when the config is missing, the layer is not added. If we break out the layer creation code to a separate function that returns `Option<Layer>` (don't know what's the exact layer type), we could add a simple test for that.